src/BuildScriptGenerator/DefaultDockerfileGenerator.cs

```diff
@@ -255,5 +261,49 @@ private string ConvertToRuntimeVersion(string platformName, string platformVersi
         {
             return this.platformDetector.GetCompatiblePlatforms(ctx);
         }
+
+        private void ValidateDockerfileImages(DockerfileProperties dockerfileProperties)
+        {
+            if (string.IsNullOrEmpty(dockerfileProperties.BuildImageName))
+            {
+                this.LogInvalidDockerfileImageException(
+                    "Provided build image name parsed from the provided --build-image argument is empty. " +
+                    "Please provide a valid value for --build-image, or remove the argument to use the default " +
+                    "'cli:stable' build image from mcr.microsoft.com/oryx.");
+            }
+
+            if (string.IsNullOrEmpty(dockerfileProperties.BuildImageTag))
+            {
+                this.LogInvalidDockerfileImageException(
+                    "Provided build image tag parsed from the provided --build-image argument is empty. " +
+                    "Please provide a valid value for --build-image, or remove the argument to use the default " +
+                    "'cli:stable' build image from mcr.microsoft.com/oryx.");
+            }
+
+            if (string.IsNullOrEmpty(dockerfileProperties.RuntimeImageName))
+            {
+                this.LogInvalidDockerfileImageException(
+                    "Either the value provided to the --runtime-platform argument is empty, or the platform " +
+                    "discovered by Oryx does not have any available or supported runtime images. Please view the following " +
+                    "document for more information on runtimes supported by Oryx: " +
+                    "https://github.com/microsoft/Oryx/blob/main/doc/supportedRuntimeVersions.md");
+            }
+
+            if (string.IsNullOrEmpty(dockerfileProperties.RuntimeImageTag))
+            {
+                this.LogInvalidDockerfileImageException(
```

Comment:
Could you please confirm that we should see these exceptions in our logs, and be able to measure how many customers got it?

Reply:
OK, I see below that we do log these before throwing.